src/python/CRABClient/CrabRestInterface.py

```diff
@@ -174,14 +174,27 @@ def makeRequest(self, uri=None, data=None, verb='GET'):
             url = url + '?' + data
 
         command = ''
-        command += 'curl -v -X {0}'.format(verb)
-        command += ' -H "User-Agent: %s/%s"' % (self['userAgent'], self['version'])
-        command += ' -H "Accept: */*"'
-        command += ' --data @%s' % path
-        command += ' --cert "%s"' % self['cert']
-        command += ' --key "%s"' % self['key']
-        command += ' --capath "%s"' % caCertPath
-        command += ' "%s" | tee /dev/stderr ' % url
+
+        # CRAB_useGoCurl env. variable is used to define how request should be executed
+        # If variable is set, then goCurl is used for command execution: https://github.com/vkuznet/gocurl
+        if os.getenv('CRAB_useGoCurl'):
+            command += '/cvmfs/cms.cern.ch/cmsmon/gocurl -verbose 2 -method {0}'.format(verb)
+            command += ' -header "User-Agent: %s/%s"' % (self['userAgent'], self['version'])
+            command += ' -header "Accept: */*"'
+            command += ' -data "%s"' % data
```

Comment:
hm, we should also pass here file, not data.

Reply:
I do not understand what you mean. If you want feedback, please elaborate. For GET there is no data, everything is in the URL. Do you refer to the file where we write the data for POST/PUT calls to avoid the problem with curl argument length ?